contracts/facets/YGFacetZaynFi.sol

```diff
@@ -235,6 +242,118 @@ contract YGFacetZaynFi is IYGFacetZaynFi {
+                    yield.withdrawnCollateral[user] = deposit;
+
+                    emit OnWithdrawnBalanceRestored(termId, user, deposit);
+                }
+
+                unchecked {
+                    ++j;
+                }
+            }
+
+            // Safety check but most likely the case
+            if (withdrawnTooMuch == 0) {
+                unchecked {
+                    ++i;
+                }
+                continue;
+            }
+            // Restore currentTotalDeposit to what it's supposed to be
+            yield.currentTotalDeposit += withdrawnTooMuch;
+
+            // We calculate the current shares we actually need in total for this term
+            uint neededShares = (yield.currentTotalDeposit * yield.totalShares) /
+                yield.totalDeposit;
+
+            // withdrawnTooMuch was withdrawn back to the smart contract, we must send it back to the yield vault
+            IZaynZapV2TakaDAO(zapAddress).zapInEth{value: withdrawnTooMuch}(vaultAddress, termId);
+
+            // Get the shares after
+            uint sharesAfter = IZaynVaultV2TakaDao(vaultAddress).balanceOf(termId);
+
+            // require(sharesAfter <= neededShares, "Too many shares for deposit!");
+            if (neededShares > sharesAfter) {
+                // If we still need more shares (which is most likely the case), we compensate by putting the missing amount into the vault
+                // Calculate the amount of eth we need to deposit to get the desired shares
+                uint pricePerShare = IZaynVaultV2TakaDao(vaultAddress).getPricePerFullShare();
+
+                uint neededEth = (15 * (neededShares - sharesAfter) * pricePerShare) / 10 ** 19; // We ask for 150% of the shares we need to compensate for the slippage
+
+                // Make sure we have enough eth
+                require(neededEth + usedValue <= msg.value, "Not enough ETH value sent");
+
+                // Deposit the amount of shares we lost
+                IZaynZapV2TakaDAO(zapAddress).zapInEth{value: neededEth}(vaultAddress, termId);
+
+                // Increment the used value so far
+                usedValue += neededEth;
+
+                // Validate the amount of shares deposited
+                sharesAfter = IZaynVaultV2TakaDao(vaultAddress).balanceOf(termId);
+
+                // If we deposited more shares than we needed, we withdraw the extra shares and send them back to the caller
+                uint withdrawnExtraEth = IZaynZapV2TakaDAO(zapAddress).zapOutETH(
+                    vaultAddress,
+                    sharesAfter - neededShares,
+                    termId
+                );
+
+                // Give the extra eth back to msg.sender
+                usedValue -= withdrawnExtraEth;
+            }
+
+            unchecked {
+                ++i;
+            }
+
+            emit OnYieldTermUpdated(termId);
```

Comment:
This one seems redundant, maybe add how much ETH was restored and how much was compensated. So add: withdrawnTooMuch, (neededEth - withdrawnExtraEth) & the final balanceOf from zaynfi